### PR TITLE
Maintenance and udpates

### LINKED ISSUE #4: assemble-weights-from-Qiita: make more efficient
Evidently 400 GB of RAM is inadequate for `assemble-weights-from-Qiita` with the SILVA reference db + various EMPO3 types.

Either this command should be more efficient, or else expose options to control memory usage. E.g., I suspect the bespoke classifier training step is where we are running out of memory... just expose `reads-per-batch`?

### PR DESCRIPTION
**Pull request by BenKaehler**
- fixed bug in `summarize-Qiita-metadata-category-and-contexts` introduced by `q2-feature-classifier` interface change
- added `allow_weight_outside_reference` to `generate_class_weights`
- added `n_jobs` and `reads_per_batch` to `summarize-Qiita-metadata-category-and-contexts` (closes #4)
- wrote a unit test for `allow_weight_outside_reference`
- updated citation
- updated README
- added many help texts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,4 +4,6 @@
 
 This is a QIIME 2 plugin. For details on QIIME 2, see https://qiime2.org.
 
-For details on this plugin, please see the [tutorial](https://forum.qiime2.org/t/using-q2-clawback-to-assemble-taxonomic-weights/5859) or the [preprint](https://doi.org/10.1101/406611).
+For details on this plugin, please see the [tutorial](https://forum.qiime2.org/t/using-q2-clawback-to-assemble-taxonomic-weights/5859) or the [article](https://doi.org/10.1038/s41467-019-12669-6).
+
+If you like the idea of using taxonomic weights to improve your taxonomic classification accuracy, but don't want to run clawback, please check out our ready-made weights at [readytowear](https://github.com/BenKaehler/readytowear).
```
Comment (nbokulich) [inline README.md:11]: maybe make this info more prominent (if this is what we plan to begin recommending to average users)

File: q2_clawback/plugin_setup.py [hunk 2]
```diff
@@ -52,7 +52,9 @@
             'reference_sequences': FeatureData[Sequence],
             'samples': FeatureTable[Frequency],
             'taxonomy_classification': FeatureData[Taxonomy]},
-    parameters={'unobserved_weight': Float, 'normalise': Bool},
+    parameters={'unobserved_weight': Float,
```
Comment (nbokulich) [inline q2_clawback/plugin_setup.py:93]: please also add parameter descriptions (and input/output descriptions for that matter!) if you plan to actually have people use this

File: q2_clawback/plugin_setup.py [hunk 3]
```diff
@@ -80,7 +82,9 @@
                 'context': Str,
                 'unobserved_weight': Float,
                 'normalise': Bool,
-                'metadata_key': Str},
+                'metadata_key': Str,
```
Comment (nbokulich) [inline q2_clawback/plugin_setup.py:145]: please also add parameter descriptions (and input/output descriptions) for everything

File: q2_clawback/tests/test_clawback.py [hunk 4]
```diff
@@ -57,13 +72,25 @@ def test_assemble_weights_from_Qiita(self):
...
-                   for s in seqs if s.metadata['id'] in tax)
-        self.assertEqual(taxa, set(weights.ids(axis='observation')))
-        self.assertEqual(weights.shape, (len(taxa), 1))
-        weights = weights.data('Weight')
-        self.assertTrue((weights > 0.).all())
-        self.assertTrue((weights < 1.).all())
-        self.assertAlmostEqual(weights.sum(), 1.)
+        self._check_weights(weights)
+
+    def test_allow_weight_outside_reference(self):
```
Comment (nbokulich) [inline q2_clawback/tests/test_clawback.py:77]: stupid test but can you add a test that confirms that the taxonomy does not match weights error is raised when appropriate? (if that test does not already exist)

**Review by nbokulich — CHANGES_REQUESTED** [review]
Looks good @BenKaehler I have some minor comments to make this more useable to others.

File: q2_clawback/tests/test_clawback.py (hunk 4 above)
Comment (BenKaehler) [reply q2_clawback/tests/test_clawback.py:77]: Already exists. Check out `assertRaises` on line 88.

File: README.md (hunk 1 above)
Comment (BenKaehler) [reply README.md:11]: I have added a heading to draw the eye.

File: q2_clawback/plugin_setup.py (hunk 2 above)
Comment (BenKaehler) [reply q2_clawback/plugin_setup.py:93]: Done.

File: q2_clawback/plugin_setup.py (hunk 3 above)
Comment (BenKaehler) [reply q2_clawback/plugin_setup.py:145]: Done.

**Review by nbokulich — APPROVED** [review]
Thanks @BenKaehler ! This is all looking great and I can confirm all is working fine offline (ran a little test to ensure that taxonomy subsets work okay with the new option)

**Pull request merged**